### DISCUSSION TITLE
Customized chart to display the state of item(s) over time with each state having an associated color.

### DISCUSSION THREAD

**Discussion by sysmith1** [Live-Charts/LiveCharts2#262] [General]
I'm trying to create a chart that is similar to the following image.  Do you have any suggestions on the type of chart that I could use or modify to produce something similar?  The control state of 5 HVAC units are logged in a database.  I need to chart the state of each unit over time.  The HVAC units can be in one of five states: ByPassOff, ByPassOn, FanOn, Stage1On, Stage2On.  Each state is represented by a different color.  Nothing is displayed when in ByPassOff state. 
The states of each of the 5 HVAC units are all the same in the example image. 
![image](https://user-images.githubusercontent.com/51513423/139114506-c5c831d8-a24b-4f02-96a2-28b655f03a6c.png)

**Comment by preintel** [comment]
You can use CartesianCharts and use a LineSeries but if I were coding this I would prefer using CartesianCharts and just adding Sections for each of the 5 HVACs and use different colors and x widths (Date/Time ticks elapsed) to represent the different states. Are you building it with Winforms or WPF or on another architecture? 

There is example code on how to build a line series and sections in the sample code section you can check out here: https://github.com/beto-rodriguez/LiveCharts2/tree/master/samples (the ViewModel is a class that builds the backbone of the chart type which the cartesianchart series/properties can be assigned from. Let us know if it is Winforms/WPF/etc so we can help you out better.

**Reply by sysmith1** [reply]
Is there a LineSeries data point with a fill color somewhat like the HeatPoint that has a weight that can map to a color for each state?

**Reply by preintel** [reply]
I am not sure, since you are using datetime on the x-axis probably need to use DateTimePoint for the LineSeries then use a mapper to change the colors of the line depending on the x-axis ranges, this way you draw the lineseries for a HVAC unit using DateTimePoint then have the mapper change the colors for parts of the series. This can work for display purposes but if you want the user to interact with the chart that will be more complicated. I program in Winforms so not too sure about WPF solutions, maybe another user can be of more help.

**Comment by sysmith1** [comment]
I am using WPF.

**Comment by sysmith1** [comment]
I started with using LiveCharts and not LiveCharts2.  Do you recommend upgrading to LiveChart2 and if so, do you have any migration instructions?

**Reply by waffle-lord** [reply]
Like, in general? Or specifically for your question?

**Generally**, it depends how much you want to fight with lv2. A lot of features are in development. So you may not be able to, or might have a hard time, getting certain things to work the way you want. That said, a lot of things already work pretty well.

I personally made the jump, because I really like how its turning out and couldn't wait.

I'm basically just putting certain things on the backburning until they are implemented.

**For your question**, you'll probably have an easier time in lv1, but I can't be certain. I've never tried to make a graph control like that.
